scipy/optimize/_constraints.py

```diff
@@ -249,6 +250,57 @@ def __init__(self, constraint, x0, sparse_jacobian=None,
         self.bounds = (lb, ub)
         self.keep_feasible = keep_feasible
 
+    def is_feasible(self, x):
+        """
+        The truth of whether a given vector of independent variables
+        satisfies the constraint.
+
+        Parameters
+        ----------
+        x : array-like
+            Vector of independent variables.
+
+        Returns
+        -------
+        feasible : bool
+            Whether the vector of independent variables satisfies the 'm'
+            constraints.
+        """
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning)
+            ev = self.fun.fun(np.asarray(x))
+        feasible_lb = np.all(ev >= self.bounds[0])
+        feasible_ub = np.all(ev <= self.bounds[1])
+
+        return feasible_lb and feasible_ub
+
+    def excess_violation(self, x):
+        """
+        How much the constraint is exceeded by.
+
+        Parameters
+        ----------
+        x : array-like
+            Vector of independent variables
+
+        Returns
+        -------
+        excess : array-like
+            How much the constraint is exceeded by, for each of the  'm'
+            constraints
+        """
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning)
+            ev = self.fun.fun(np.asarray(x))
+
+        excess_lb = self.bounds[0] - ev
```

Comment:
Not really important, but `np.maximum` looks useful here `excess_lb = np.maximum(self.bounds[0] - ev, 0)`.

Reply:
Good catch